### ISSUE TITLE
Cannot remove local cluster (in-cluster)

### ISSUE THREAD

**Issue by morremeyer** [argoproj/argo-cd#14531]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

The local cluster (in-cluster) can't be removed.

```sh
❯ argocd cluster rm in-cluster
Are you sure you want to remove 'in-cluster'? Any Apps deploying to this cluster will go to health status Unknown.[y/n] y
FATA[0001] rpc error: code = NotFound desc = cluster "https://kubernetes.default.svc" not found 
```

I think this might be due to the case that it's a "magic" cluster that is not loaded from a secret?

**To Reproduce**

```sh
argocd cluster rm in-cluster
```

**Expected behavior**

Local cluster is removed from clusters.

**Version**

```shell
❯ argocd version
argocd: v2.7.7+4650bb2.dirty
  BuildDate: 2023-07-05T22:18:46Z
  GitCommit: 4650bb2817c3c81405f40cf77e93ef2b5fb275fb
  GitTreeState: dirty
  GoVersion: go1.20.5
  Compiler: gc
  Platform: darwin/arm64
argocd-server: v2.7.7+4650bb2.dirty
  BuildDate: 2023-07-05T19:42:32Z
  GitCommit: 4650bb2817c3c81405f40cf77e93ef2b5fb275fb
  GitTreeState: dirty
  GoVersion: go1.19.10
  Compiler: gc
  Platform: linux/amd64
  Kustomize Version: v5.0.1 2023-03-14T01:32:48Z
  Helm Version: v3.11.2+g912ebc1
  Kubectl Version: v0.24.2
  Jsonnet Version: v0.19.1
```

**Logs**

Logs from `argocd-server` pod:

```
time="2023-07-15T20:31:04Z" level=info msg="received unary call /cluster.ClusterService/Delete" grpc.method=Delete grpc.request.content="id:<value:\"https://kubernetes.default.svc\" > " g
rpc.service=cluster.ClusterService grpc.start_time="2023-07-15T20:31:04Z" span.kind=server system=grpc
time="2023-07-15T20:31:04Z" level=info msg="finished unary call with code NotFound" error="rpc error: code = NotFound desc = cluster \"https://kubernetes.default.svc\" not found" grpc.cod
e=NotFound grpc.method=Delete grpc.service=cluster.ClusterService grpc.start_time="2023-07-15T20:31:04Z" grpc.time_ms=9.232 span.kind=server system=grpc
```

**Comment by yyzxw** [comment]
The local cluster is the host cluster where argoCD runs and cannot be deleted.

**Comment by morremeyer** [comment]
I'm aware that it is the cluster where ArgoCD runs.

I reported this as a bug for two reasons:

* There is no documentation that not being able to delete it is a design decision, leading me to believe that that is not per design
* If the cluster that ArgoCD runs in does not have any resources that should be managed by ArgoCD, following the principle of least privilege, it should be deletable.

**Comment by crenshaw-dev** [comment]
@morremeyer it is possible to disable in-cluster access: https://github.com/argoproj/argo-cd/blob/e10c4186a5765c10c061a6d7d8f46c8e8b381c9e/docs/operator-manual/argocd-cm.yaml#L316-L317

We should 1) document this better and 2) have the CLI suggest changing that setting if someone tries to delete in-cluster.

**Comment by morremeyer** [comment]
I didn't find a good place to add a docs hint, since there is no documentation yet about cluster management in the CLI (besides the actual command documentation).

I don't think this should be added in https://github.com/argoproj/argo-cd/blob/f1607fee7c3bfad8eb56a53130cf9ac7cc22dd34/cmd/argocd/commands/cluster.go#L45, so I'm happy about any ideas anyone might have.

**Comment by gdhameeja** [comment]
If this is still open, maybe I can take this up.

**Comment by morremeyer** [comment]
Added a first documentation PR, and let's just not discuss my definition of "next Friday", yeah? 😁

**Comment by morremeyer** [comment]
This is now documented, and since I won't get to work on it in the foreseeable future, I closed #16556.

Happy to keep this open or close it, I don't have a preference either way.

**Comment by agaudreault** [comment]
Closing since it is documented
